packages/backend/src/payment-method/ilp/service.ts

```diff
@@ -217,12 +219,27 @@ async function pay(
     })
   }
 
+  if (!outgoingPayment.quote.ilpQuoteDetails) {
+    outgoingPayment.quote.ilpQuoteDetails =
+      await deps.ilpQuoteDetailsService.getByQuoteId(outgoingPayment.quote.id)
+
+    if (!outgoingPayment.quote.ilpQuoteDetails) {
+      throw new PaymentMethodHandlerError(
+        'Could not find required ILP Quote Details',
+        {
+          description: 'ILP Quote Details not found',
+          retryable: false
+        }
+      )
+    }
+  }
+
```

Comment:
This is how im ensuring we always have the quote details. I also considered simply ensuring the caller passes in a `quote` with `ilpQuoteDetails`, but since the interface is generic (ie same for ILP and LOCAL) I think we'd still need to add a check like this to make typescript happy, so I think adding this and the ilpQuoteDetailsService as a dep is required either way.